File: app/src/main/java/pl/revanmj/smspasswordnotifier/data/WhitelistViewModel.java
package pl.revanmj.smspasswordnotifier.data;

import android.app.Application;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;

import java.util.List;

public class WhitelistViewModel extends AndroidViewModel {
    private WhitelistRepository mRepository;
    private LiveData<List<WhitelistItem>> mWhitelist;

    public WhitelistViewModel(Application application) {
        super(application);
        mRepository = new WhitelistRepository(application);
        mWhitelist = mRepository.getWhitelist();
    }

    public LiveData<List<WhitelistItem>> getWhitelist() { return mWhitelist; }

    public void insert(WhitelistItem... items) { mRepository.insert(items); }

    public void delete(WhitelistItem... items) { mRepository.delete(items); }

    public WhitelistItem getItemByName(String name) { return mRepository.getItemByName(name); }
}

File: app/src/main/java/pl/revanmj/smspasswordnotifier/data/WhitelistViewModel.kt
package pl.revanmj.smspasswordnotifier.data

import android.app.Application

import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData

class WhitelistViewModel(application: Application) : AndroidViewModel(application) {
    private val mRepository: WhitelistRepository = WhitelistRepository(application)
    val whitelist: LiveData<List<WhitelistItem>>

    init {
        whitelist = mRepository.whitelist
    }

    fun insert(vararg items: WhitelistItem) {
        mRepository.insert(*items)
    }

    fun update(vararg items: WhitelistItem) {
        mRepository.update(*items)
    }

    fun delete(vararg items: WhitelistItem) {
        mRepository.delete(*items)
    }

    fun getItemByName(name: String): WhitelistItem {
        return mRepository.getItemByName(name)
    }
}
